### PR TITLE
TD Balance Changes. 20172606

### PR DESCRIPTION
**Pull request by AoAGeneral**
The list of changes do not affect the new hitboxes of the structures. The engineer external capture was dropped currently due to the new change. Same applies to the MCV price change due to some of these changes. These changes will be looked at in the playtest and if need be after the next release.

+ Orca damage increased from 25 to 28.
Damage is increased because of their lack of damage vs armored units and structures.

+ Apache ammo reload reduced from 200 to 40.
Apaches gaining a full clip of ammo is a bit to strong. Firing the first shot starts the timer tick and will reload even if the clip is not fully reloaded. This also fixes so they don't sit on top of infantry and blow them all away to quickly.

+ Apache ammo reload amount reduced from 10 to 1.
Fixed in unison with ammo reload timer.

+ Sam now fires two missiles. (CNC95)
Does extra damage.

+ Sam range increase from 8c0 to 10c0.
Compensate for the aircraft changes. Also had a problem of MSAMs were always the better choice rather then the structure SAMs itself. This change will make them more viable.

+ Sam range limit from 9c614 to 12c0.
Makes it so aircraft can still escape while the missile won't detonate at the edge of the circle range.

+ Humvee damag vs light increased from 50 to 55.
Compensates for the price of the hummer. Having just the HP wasn't enough to counter certian builds of Nod. Enabling them a slightly higher damage allows them to kill buggies and bikes a bit more effectively. Makes them also decent arty and MRLS killers.

+ Crates got some love.

+ Viceroids now spawn.

+ An atomic bomb can now detonate. (CNC95)

+ Crate veterancy level now gives general rank instead of stripe 1.
This makes grabbing crates all the more rewarding and riskier. Such as willing to bring a mammoth tank or commando to the crate rather then a minigunner. Instead of a crate granting level 1 for units it will now give rank level 4 promoting to general status.

### CODE REVIEW

**Comment by GraionDilach** [comment]
I'd reword the last paragraph. I thought you were introducing a veterancy level above heroic and took me a while to realize that you meant you set the veterancy crate to instantly promote to heroic.

**Comment by AoAGeneral** [comment]
Done. Thanks!

**Comment by pchote** [comment]
> Firing the first shot starts the timer tick and will reload even if the clip is not fully reloaded.

What if we fixed this instead? (note: would also affect orca).

File: mods/cnc/rules/structures.yaml [hunk 1]
```diff
@@ -803,7 +803,7 @@ SAM:
 	WithTurretedSpriteBody:
 	AutoSelectionSize:
 	Armament:
-		Weapon: SAMMissile
+		Weapon: Dragon
```
Comment (pchote) [inline mods/cnc/rules/structures.yaml]: This now leaves the `SAMMissile` counterintuitively only used by the advanced guard tower.  Please rename it to `TowerAAMissile` if this change is intentional.

**Comment by pchote** [comment]
Making it not start the reload timer until it runs out of ammo, which is the expected behaviour (consistent with other units like the rocket launcher, and with reality: if you're swapping a magazine you need to wait until you are done with the old one).

**Comment by AoAGeneral** [comment]
If its unexpected behavior and you want to fix it I say go for it. I can always drop the change for now and rework it around with the new change.

File: mods/cnc/rules/structures.yaml (hunk 1 above)
Comment (AoAGeneral) [inline mods/cnc/rules/structures.yaml]: SAMMissile is now TowerAAMissile.

**Comment by reaperrr** [comment]
> If its unexpected behavior and you want to fix it I say go for it. I can always drop the change for now and rework it around with the new change.

Unless I'm misunderstanding something here, that's not a bug, you just need to set `ResetOnFire: true` on the AmmoPool.

File: mods/cnc/weapons/smallcaliber.yaml [hunk 2]
```diff
@@ -114,6 +114,17 @@ MachineGun:
 	Warhead@2Eff: CreateEffect
 		Explosions: piffs
 
+MachineGunH:
```
Comment (ltem) [inline mods/cnc/weapons/smallcaliber.yaml]: Probably I'm wrong but I cannot find where `MachineGunH` is actually used (Humvee?).

**Review by ltem — COMMENTED** [review]
I'm fine with all the changes and I trust your balance experience and after reaperrr's and my comments are clarified I would approve to this PR.

File: mods/cnc/rules/misc.yaml [hunk 3]
```diff
@@ -22,8 +22,18 @@ CRATE:
 		Condition: cloak-crate-collected
 	GiveMcvCrateAction:
 		SelectionShares: 0
-		NoBaseSelectionShares: 120
+		NoBaseSelectionShares: 50
 		Units: mcv
+	ExplodeCrateAction:
```
Comment (ltem) [inline mods/cnc/rules/misc.yaml]: The nuke won't damage the unit, which triggered the crate/explosion. That's unexpected.

Comment (reaperrr) [inline mods/cnc/rules/misc.yaml]: The Atomic weapon's SpreadDamage warheads need "AffectsParent: true" to fix that.

File: mods/cnc/weapons/smallcaliber.yaml (hunk 2 above)
Comment (AoAGeneral) [inline mods/cnc/weapons/smallcaliber.yaml]: Can't believe I missed that. Fixed. Its now used on the JEEP/Humvee

File: mods/cnc/rules/misc.yaml (hunk 3 above)
Comment (AoAGeneral) [inline mods/cnc/rules/misc.yaml]: Should that be added in this PR or a seperate PR?

Comment (GraionDilach) [inline mods/cnc/rules/misc.yaml]: Note that this makes the Temple of Nod affected/destroyable by it's own nuke - which I consider a good thing. I'd check and add the same tag to the Ion Cannon if missing for consistency.

File: mods/cnc/weapons/smallcaliber.yaml (hunk 2 above)
Comment (ltem) [inline mods/cnc/weapons/smallcaliber.yaml]: Optional: You can even reduce the amount of lines by using the inheritance feature.  
```yaml
MachineGunH:
  Inherits: MachineGun
  Warhead@1Dam: SpreadDamage
    Versus:
      Light: 55
```

File: mods/cnc/rules/misc.yaml (hunk 3 above)
Comment (AoAGeneral) [inline mods/cnc/rules/misc.yaml]: Added and done.

**Review by ltem — APPROVED** [review]
All requests are fixed. So :+1: from me.

Maybe unnecessary but to clarify the aircraft reload situation (I'm fine how it is done now).
Right now the reload counter of Apache (and Orca) starts as soon as the magazine isn't full and won't be interrupted/reset by additional fire-sequences. The `ResetOnFire: true` setting would reset the reload counter with every shot, delaying the reload in different way as by modifying `ReloadCount` or `SelfReloadDelay`.

**Comment by AoAGeneral** [comment]
Good to know for the future then. I will play around with these in the future. Currently I want to fix the mechanic of how Apache works while focusing on the units vs new hitboxes on the structures.

**Review by reaperrr — APPROVED** [review]

File: mods/cnc/weapons/smallcaliber.yaml (hunk 2 above)
Comment (reaperrr) [inline mods/cnc/weapons/smallcaliber.yaml]: I fixed that before merging.

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d677e7714298fdbb10ef7302d286786734ce3d5b...ac41ac10ec820b6ede69bde676a39d5a2e6e2a53)
